api/app/Http/Controllers/Api/AuthController.php

```diff
@@ -77,4 +81,75 @@ public function register(UserRegistrationModel $request)
 
         User::create($account);
     }
+
+    /**
+     * Send a mail for resetting the password.
+     *
+     * @param \App\Http\Requests\Api\PasswordResetModel $request
+     *
+     * @return mixed
+     */
+    public function sendResetMail(PasswordResetModel $request)
+    {
+        $user_email = $request->email;
+        $user = User::where('email', $user_email)->first();
+        if ($user && $this->notSpamming($user, env('PASSWORD_RESET_MINUTES', 30))) {
+            // User exists and had no request < PASSWORD_RESET_MINUTES
+            $uuid_token = Uuid::generate(4);
+            $url = url('/reset/'.$uuid_token);
+            $request->request->add(['url' => $url]);
+            $request->request->add(['name' => $user->name]);
+            $this->sendPasswordResetMail($request);
+            // Store in database
+            $data = [
+                'user_id' => $user->id,
+                'token' => $uuid_token,
+                'created_at' => date('Y-m-d H:i:s'),
+            ];
+            PasswordReset::create($data);
+        }
+    }
+
+    private function notSpamming(User $user, $minutes)
+    {
+        if ($user) {
+            $last_password_request = PasswordReset::where('user_id', $user->id)->orderBy('created_at', 'desc')->first();
+            if (! $last_password_request) {
+                // No request were stored yet
+                return true;
+            }
+            $last_password_request_time = $last_password_request->created_at;
+            $this->isInsideInterval($last_password_request_time, $minutes);
+        }
+
+        return false;
+    }
+
+    private function isInsideInterval($lastTime, $minutes)
```

Comment:
You can just use the Carbon package that is in Laravel that is fully tested

http://carbon.nesbot.com/docs/#api-comparison